File: tui.js
#!/usr/bin/env node
/**
 * Root entry point — launches the TUI from src/JavaScript/tui.js.
 *
 * Run from repo root:
 *   node tui.js
 */

const path = require('path');
const fs = require('fs');

const tuiScript = path.join(__dirname, 'src', 'JavaScript', 'tui.js');

if (!fs.existsSync(tuiScript)) {
  console.error(`ERROR: TUI script not found at ${tuiScript}`);
  process.exit(1);
}

// Change to src/JavaScript so relative requires work
process.chdir(path.join(__dirname, 'src', 'JavaScript'));

// Load and execute tui.js
require(tuiScript);

File: tui.py
#!/usr/bin/env python3
"""Root entry point — launches the TUI from src/Python/tui.py.

Run from repo root:
    python tui.py
"""

import os
import sys
from pathlib import Path

_repo_root = Path(__file__).parent
_tui_script = _repo_root / "src" / "Python" / "tui.py"

if not _tui_script.exists():
    print(f"ERROR: TUI script not found at {_tui_script}")
    sys.exit(1)

# Add src/Python to sys.path so tui.py can import helpers/config
sys.path.insert(0, str(_repo_root / "src" / "Python"))
os.chdir(str(_repo_root / "src" / "Python"))

# Execute tui.py
exec(open(_tui_script, encoding="utf-8").read())
